iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractTapChanger.java

```diff
@@ -43,21 +44,22 @@
 
     protected final TDoubleArrayList targetDeadband;
 
-    protected AbstractTapChanger(Ref<? extends VariantManagerHolder> network, H parent,
+    protected AbstractTapChanger(Ref<? extends VariantManagerHolder> networkRef, H parent,
                                  int lowTapPosition, List<S> steps, TerminalExt regulationTerminal,
-                                 int tapPosition, boolean regulating, double targetDeadband, String type) {
-        this.network = network;
+                                 Integer tapPosition, boolean regulating, double targetDeadband, String type) {
+        this.networkRef = networkRef;
         this.parent = parent;
+        network = parent.getNetwork();
         this.lowTapPosition = lowTapPosition;
         this.steps = steps;
         steps.forEach(s -> s.setParent(this));
         this.regulationTerminal = regulationTerminal;
-        int variantArraySize = network.get().getVariantManager().getVariantArraySize();
+        int variantArraySize = networkRef.get().getVariantManager().getVariantArraySize();
         this.tapPosition = new TIntArrayList(variantArraySize);
         this.regulating = new TBooleanArrayList(variantArraySize);
         this.targetDeadband = new TDoubleArrayList(variantArraySize);
         for (int i = 0; i < variantArraySize; i++) {
-            this.tapPosition.add(tapPosition);
+            this.tapPosition.add(tapPosition != null ? tapPosition : Integer.MIN_VALUE);
```

Comment:
Maybe it would be more clear to just replace the TIntArray by a List<Integer> and performance loss will be neglictable?